diff --git a/dist/node/module-runner.d.ts b/dist/node/module-runner.d.ts
index 511f6af6d5cbc0e12aedfd393017e85c80386a68..fd7fc31112e31d48f98bc7e00861d2197f655ca0 100644
--- a/dist/node/module-runner.d.ts
+++ b/dist/node/module-runner.d.ts
@@ -158,7 +158,7 @@ interface InterceptorOptions {
 //#region src/module-runner/types.d.ts
 interface ModuleRunnerImportMeta extends ImportMeta {
   url: string;
-  env: ImportMetaEnv;
+  env: ImportMetaEnv & (ImportMeta extends { readonly env: infer Environment } ? Environment : {});
   hot?: ViteHotContext;
   [key: string]: any;
 }
